### PR TITLE
Feature/no replacevars include

### PR DESCRIPTION
**Pull request by StaticRocket**
Remove the need for a replacevars file by placing all substitutions in the rst_prolog. This restructures things just a little bit so we can get access to required variables before we enter the application context.

It also removes some of the errors we've seen that seem to break the global include file and propagate to other files. I split the commit to make this easier to review. Let me know if you believe these should be squashed before it's merged.

### CODE REVIEW

**Comment by StaticRocket** [comment]
This will uncover other document errors. That's the whole point. This just means I was right.

@Siddharth-Vadapalli-at-TI this should help address the issue you were seeing.

File: conf.py [hunk 1]
```diff
@@ -273,10 +273,11 @@
...
 family_config_inputfile = f"{FAMILY}/{FAMILY}_{OS}_config.txt"
 
-# Hash table for Replacement Variables
-family_replacevars = {}
-# Hash table for Configuration Values
-family_configvals = {}
+# Hash table for Replacement Variables and Config Values
+family_replacevars, family_configvals = interpretvalues.read_familyvals(family_config_inputfile)
+
+# Unpack replacement variables and place them in the preable for all documents
```
Comment (StaticRocket) [inline conf.py:279]: `preable` - Just caught this typo, will address with whatever other comments this gets

**Review by makohoek — COMMENTED** [review]
When building with:

```
$ make DEVFAMILY=AM62PX OS=android
```

A new warning is observed:
```
reading sources... [100%] devices/AM62PX/android/index                                                                                                                                                                                                                                                                                                                                                                                      
/mnt/work/ti/src/processor-sdk-doc/source/devices/AM62PX/android/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```

**Comment by StaticRocket** [comment]
> /mnt/work/ti/src/processor-sdk-doc/source/devices/AM62PX/android/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.

That is a valid warning. The tag on that page was invalid. It needs a newline between it and the section header.

**Review by makohoek — APPROVED** [review]
Build tested both Android configs:
```
$ make DEVFAMILY=AM62PX OS=android
$ make DEVFAMILY=AM62X OS=android
```

Also reviewed the changes. Looks good to me!

**Comment by StaticRocket** [comment]
@cshilwant this dramatically decreases the number of `AttributeError` warnings CI is catching right now.

**Comment by cshilwant** [comment]
Screenshot_20241205_092703_Chrome

@StaticRocket you might have to rebase. The changes looks fine

**Comment by StaticRocket** [comment]
@cshilwant conflict resolved

**Review by cshilwant — APPROVED** [review]

**Review by praneethbajjuri — APPROVED** [review]

**Pull request merged**